libtenzir/builtins/formats/syslog.cpp

```diff
@@ -10,41 +10,156 @@
 #include <tenzir/format/reader_factory.hpp>
 #include <tenzir/format/syslog.hpp>
 #include <tenzir/plugin.hpp>
+#include <tenzir/series_builder.hpp>
 #include <tenzir/table_slice_builder.hpp>
 #include <tenzir/to_lines.hpp>
 
 namespace tenzir::plugins::syslog {
 
 namespace {
 
+struct syslog_builder {
+public:
+  using message_type = format::syslog::message;
+
+  syslog_builder(const type& ty) : builder_(ty) {
+  }
```

Comment:
Unless I'm missing something: Are the types passed to all three constructors not effectively constant? If so, I would just remove this parameter and use the types directly in these classes. This would also simplify the initialization code below.